[CI] ModelRegistryIT testGetModel failing

**Build Scans:**
- [elasticsearch-periodic #3691 / release-tests](https://gradle-enterprise.elastic.co/s/ygggd4kxjqyqa)
- [elasticsearch-pull-request #27735 / release-tests](https://gradle-enterprise.elastic.co/s/e7ftsqkff4k2k)
- [elasticsearch-pull-request #27711 / release-tests](https://gradle-enterprise.elastic.co/s/cwqfmqnhytd6m)
- [elasticsearch-periodic #3678 / release-tests](https://gradle-enterprise.elastic.co/s/p3xkwdmsaw6oy)
- [elasticsearch-periodic #3662 / release-tests](https://gradle-enterprise.elastic.co/s/vrqdlk5w2qibq)
- [elasticsearch-periodic #3658 / release-tests](https://gradle-enterprise.elastic.co/s/mm3lqj2fjpua6)
- [elasticsearch-periodic #3654 / release-tests](https://gradle-enterprise.elastic.co/s/qgcmllfk7ugbs)
- [elasticsearch-periodic #3634 / release-tests](https://gradle-enterprise.elastic.co/s/mnss2jlntgi7w)
- [elasticsearch-periodic #3622 / release-tests](https://gradle-enterprise.elastic.co/s/6hwgrjnymdyaa)
- [elasticsearch-periodic #3618 / release-tests](https://gradle-enterprise.elastic.co/s/wucqs75d3hbd4)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:inference:internalClusterTest" --tests "org.elasticsearch.xpack.inference.integration.ModelRegistryIT.testGetModel" -Dtests.seed=4C6A62843DCC04AE -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dtests.locale=ar-QA -Dtests.timezone=America/Porto_Velho -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testGetModel'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.inference.integration.ModelRegistryIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: expected:<org.elasticsearch.xpack.inference.services.elser.ElserInternalModel@76ff58fb> but was:<org.elasticsearch.xpack.inference.services.elser.ElserInternalModel@3186e8a6>
```

**Issue Reasons:**
- [main] 12 failures in test testGetModel (1.5% fail rate in 791 executions)
- [main] 12 failures in step release-tests (70.6% fail rate in 17 executions)
- [main] 10 failures in pipeline elasticsearch-periodic (45.5% fail rate in 22 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (1.2% fail rate in 168 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -113,9 +113,6 @@ tests:
 - class: org.elasticsearch.xpack.test.rest.XPackRestIT
   method: test {p0=rollup/security_tests/Index-based access}
   issue: https://github.com/elastic/elasticsearch/issues/111631
-- class: org.elasticsearch.xpack.inference.integration.ModelRegistryIT
-  method: testGetModel
-  issue: https://github.com/elastic/elasticsearch/issues/111570
 - class: org.elasticsearch.tdigest.ComparisonTests
   method: testSparseGaussianDistribution
   issue: https://github.com/elastic/elasticsearch/issues/111721

